### ISSUE TITLE
Variables replacement does not work in nested templates

### ISSUE THREAD

**Issue by ssideleau** [serverless/serverless#3615]
# This is a Bug Report

## Description

For bug reports:
* What went wrong?

Variables replacement did not work in my resources file.
This is not a problem in 1.12.1

Using the `sls package` command the DynamoTable part of the `cloudformation-template-update-stack.json` is :
```
"DynamoTable": {
    "Type": "AWS::DynamoDB::Table",
    "Properties": {
      "TableName": "${self:service}-${self:custom.stage}-myAttribute",
      "AttributeDefinitions": [
        {
          "AttributeName": "myAttribute",
          "AttributeType": "S"
        }
      ],
      "KeySchema": [
        {
          "AttributeName": "myAttribute",
          "KeyType": "HASH"
        }
      ],
      "ProvisionedThroughput": {
        "ReadCapacityUnits": 10,
        "WriteCapacityUnits": 10
      }
    }
  }
```
* What did you expect should have happened?

The DynamoTable part of `cloudformation-template-update-stack.json` should be:
```
    "DynamoTable": {
      "Type": "AWS::DynamoDB::Table",
      "Properties": {
        "TableName": "sls-variables-test-dev-myAttribute",
        "AttributeDefinitions": [
          {
            "AttributeName": "myAttribute",
            "AttributeType": "S"
          }
        ],
        "KeySchema": [
          {
            "AttributeName": "myAttribute",
            "KeyType": "HASH"
          }
        ],
        "ProvisionedThroughput": {
          "ReadCapacityUnits": 10,
          "WriteCapacityUnits": 10
        }
      }
    }
```
* What was the config you used?

**Project Structure**
```
.
├── handler.js
├── serverless.yml
└── slsConfig
    └── aws-resources
        ├── dev.resources.yml
        └── templates
            └── dynamoDB.resources.yml
```
**serverless.yml**
```
service: sls-variables-test

custom:
  stage: ${opt:stage, self:provider.stage}
provider:
  name: aws
  runtime: nodejs6.10
  stage: dev

functions:
  hello:
    handler: handler.hello

resources:
  Resources:
    ${file(slsConfig/aws-resources/${self:custom.stage}.resources.yml)}
```
**slsConfig/aws-resources/dev.resources.yml**
```
${file(slsConfig/aws-resources/templates/dynamoDB.resources.yml)}
```
**slsConfig/aws-resources/templates/dynamoDB.resources.yml**
```
DynamoTable:
  Type: AWS::DynamoDB::Table
  Properties:
    TableName: '${self:service}-${self:custom.stage}-myAttribute'
    AttributeDefinitions:
      - AttributeName: myAttribute
        AttributeType: S
    KeySchema:
      - AttributeName: myAttribute
        KeyType: HASH
    ProvisionedThroughput:
      ReadCapacityUnits: 10
      WriteCapacityUnits: 10
```
* What stacktrace or error message from your provider did you see?

```
..........Serverless: Deployment failed!

  Serverless Error ---------------------------------------

     An error occurred while provisioning your stack: DynamoTable
     - 1 validation error detected: Value '${self:service}-${self:custom.stage}-myAttribute'
     at 'tableName' failed to satisfy constraint: Member
     must satisfy regular expression pattern: [a-zA-Z0-9_.-]+.

  Get Support --------------------------------------------
     Docs:          docs.serverless.com
     Bugs:          github.com/serverless/serverless/issues
     Forums:        forum.serverless.com
     Chat:          gitter.im/serverless/serverless

  Your Environment Information -----------------------------
     OS:                 linux
     Node Version:       6.10.2
     Serverless Version: 1.13.0
```

Similar or dependent issues:

## Additional Data

* ***Serverless Framework Version you're using***: 1.13.0
* ***Operating System***: Gnome Ubuntu 16.04
* ***Stack Trace***:
* ***Provider Error messages***:

**Comment by NaraGitHub** [comment]
I'm having this issue too

**Comment by ssideleau** [comment]
This issue seems to be fixed in serverless version 1.13.2 see https://github.com/serverless/serverless/pull/3604

**Comment by pmuens** [comment]
Thanks for pointing in that direction @ssideleau 👍 Can anyone confirm that it's fixed in `v1.13.2`?

**Comment by vyh** [comment]
This isn't exactly the error that I get in 1.13.0, but I believe it's the same issue (I pull a function's events from another file; the events use a variable from self:custom to define the responses; this works below 1.13 but in 1.13 sls deploy throws an error saying "Response config must be provided as an object."). If I put all the event definitions directly in serverless.yml, it deploys.

However, in 1.13.1 and up, I get a js error (out of memory - this or related to it?: https://github.com/serverless/serverless/issues/3627). So, I can't tell if the first issue is fixed.

**Comment by pmuens** [comment]
Thanks for responding back @vyh 

🤔 hmm, yes, it seems like the root cause is "fixed", but now the memory error pops up.

It seems like #3604 fixed the root cause for this bug. I'll close this one for now and we can continue the discussion about the memory issue in #3627.

Feel free to comment / re-open if this bug as described in the description is still around.

**Comment by NaraGitHub** [comment]
Fixed and working - Tested with 1.14.0

**Comment by pmuens** [comment]
Thanks for confirming @NaraGitHub 👍
